Source/ARTMessage.m

```diff
@@ -45,3 +49,73 @@ - (NSInteger)messageSize {
 }
 
 @end
+
+@implementation ARTMessage (Decoding)
+
++ (instancetype)fromEncodedJsonObject:(NSDictionary *)json channelOptions:(ARTChannelOptions *)options error:(NSError **)error {
```

Comment:
Definitely, but that would be changes in `ARTDataEncoder/decode:identifier:encoding:` method tweaking which wasn't the purpose of this PR. I've created a separate issue for that - https://github.com/ably/ably-cocoa/issues/1317

Reply:
But the idea of this PR was to add the `fromEncoded(…)` methods, so if there are things that the spec doesn't address about how these methods should be implemented, would now not be a good time to get those things added to the spec and then incorporate them into these methods we add?

Reply:
(@QuintinWillison thoughts?)

Reply:
I agree with @lawrence-forooghian that if we're introducing a new API then we should try to ensure we get it right from the outset. However - perhaps more in line with @maratal's thinking - as far as I can tell, the method signature is not going to need to change - regardless of how it's implemented (i.e. there's already an `NSError *` return path). So, perhaps, we could defer enhancing the implementation to match up to a spec addition to a later unit of work (i.e. another pull request)? This PR is getting pretty old and it would be nice to stay agile - I believe it's taking the code from one working state to another, right?

Reply:
@QuintinWillison @lawrence-forooghian the implementation of these new methods probably will not change either. I's `ARTDataEncoder/decode:identifier:encoding:` which is going to be changed.